### ISSUE TITLE
posix_net_conf.sh eth0 not running when scylla cluster up

### ISSUE THREAD

**Issue by nirmaayan** [scylladb/scylla#1406]
Scylla version (or git commit hash): 1.2
Cluster size: 3 
OS (RHEL/CentOS/Ubuntu/AWS AMI): i2.8xlarge

3 node cluster
When the cluster is up

> $ cat /sys/class/net/eth0/queues/*/rps_cpus
00000000,00000000,00000000,00000000
00000000,00000000,00000000,00000000

After activate

> $ sudo /usr/lib/scylla/posix_net_conf.sh eth0
00000000,00000000,00000000,00fe00fe
00000000,00000000,00000000,ff00ff00

Also after restart Scylla the results remain to be all 0

**Comment by slivne** [comment]
I am not sure this is correct

```
[centos@ip-10-181-64-140 ~]$ cat /etc/sysconfig/scylla-server 
# choose following mode: virtio, dpdk, posix
NETWORK_MODE=posix

# tap device name(virtio)
TAP=tap0

# bridge device name (virtio)
BRIDGE=virbr0

# ethernet device name
IFNAME=eth0

# setup NIC's interrupts, RPS, XPS (posix)
SET_NIC=yes

# ethernet device driver (dpdk)
ETHDRV=

# ethernet device PCI ID (dpdk)
ETHPCIID=

# number of hugepages
NR_HUGEPAGES=64

# user for process (must be root for dpdk)
USER=scylla

# group for process
GROUP=scylla

# scylla home dir
SCYLLA_HOME=/var/lib/scylla

# scylla config dir
SCYLLA_CONF=/etc/scylla

# scylla arguments (for posix mode)
SCYLLA_ARGS="--log-to-syslog 1 --log-to-stdout 0 --default-log-level info --collectd-address=127.0.0.1:25826 --collectd=1 --collectd-poll-period 3000 --network-stack posix"

## scylla arguments (for dpdk mode)
#SCYLLA_ARGS="--log-to-syslog 1 --log-to-stdout 0 --default-log-level info --collectd-address=127.0.0.1:25826 --collectd=1 --collectd-poll-period 3000 --network-stack native --dpdk-pmd"

# setup as AMI instance
AMI=yes
```

this means that scylla_prepare prio to starting scylla will run the posix_net_conf

scylla_prepare will run

```
    if [ "$SET_NIC" = "yes" ]; then
        NRXQ=`find /sys/class/net/$IFNAME/queues -name "rx-*"|wc -l`
        if [ $NRXQ -gt 1 ]; then
            CONF_ARGS=-mq
        else
            CONF_ARGS=-sq
        fi
        /usr/lib/scylla/posix_net_conf.sh $IFNAME $CONF_ARGS
```

since we get a result of 1 we will run posix_net_conf with -sq

which will cause all interrupts to be pinned to core 0 (what we want)

Could it be that you did this prior to scylla starting up - and did not wait for it to start


I tested this with scylla 1.2 **(ami-928a4dff)**

**Comment by nirmaayan** [comment]
When I test it on 1 node it didn't happened 
When I tested it on 3 node cluster it happened

**Comment by slivne** [comment]
Thanks - I'll check again on a 3 node cluster 
Is there anything else that you did that is missing here - 

are you using the AMI or installing scylla on your own ?

**Comment by nirmaayan** [comment]
Using AMI (create with ansible).
I test the $ cat /sys/class/net/eth0/queues/*/rps_cpus couple of times

**Comment by vladzcloudius** [comment]
On Jun 29, 2016 09:32, "Shlomi Livne" <notifications@github.com> wrote:
>
> I am not sure this is correct
>
> [centos@ip-10-181-64-140 ~]$ cat /etc/sysconfig/scylla-server
> # choose following mode: virtio, dpdk, posix
> NETWORK_MODE=posix
>
> # tap device name(virtio)
> TAP=tap0
>
> # bridge device name (virtio)
> BRIDGE=virbr0
>
> # ethernet device name
> IFNAME=eth0
>
> # setup NIC's interrupts, RPS, XPS (posix)
> SET_NIC=yes
>
> # ethernet device driver (dpdk)
> ETHDRV=
>
> # ethernet device PCI ID (dpdk)
> ETHPCIID=
>
> # number of hugepages
> NR_HUGEPAGES=64
>
> # user for process (must be root for dpdk)
> USER=scylla
>
> # group for process
> GROUP=scylla
>
> # scylla home dir
> SCYLLA_HOME=/var/lib/scylla
>
> # scylla config dir
> SCYLLA_CONF=/etc/scylla
>
> # scylla arguments (for posix mode)
> SCYLLA_ARGS="--log-to-syslog 1 --log-to-stdout 0 --default-log-level info
--collectd-address=127.0.0.1:25826 --collectd=1 --collectd-poll-period 3000
--network-stack posix"
>
> ## scylla arguments (for dpdk mode)
> #SCYLLA_ARGS="--log-to-syslog 1 --log-to-stdout 0 --default-log-level
info --collectd-address=127.0.0.1:25826 --collectd=1 --collectd-poll-period
3000 --network-stack native --dpdk-pmd"
>
> # setup as AMI instance
> AMI=yes
>
> this means that scylla_prepare prio to starting scylla will run the
posix_net_conf
>
> scylla_prepare will run
>
>     if [ "$SET_NIC" = "yes" ]; then
>         NRXQ=`find /sys/class/net/$IFNAME/queues -name "rx-*"|wc -l`
>         if [ $NRXQ -gt 1 ]; then
>             CONF_ARGS=-mq
>         else
>             CONF_ARGS=-sq
>         fi
>         /usr/lib/scylla/posix_net_conf.sh $IFNAME $CONF_ARGS
>
> since we get a result of 1 we will run posix_net_conf with -sq

The logic above is bogus: -sq mode should be started not only when the
result above is 1 (for Nir it would be 2, by the way). See the
corresponding logic in the posix_net_conf.sh script.

>
> which will cause all interrupts to be pinned to core 0 (what we want)
>
> Could it be that you did this prior to scylla starting up - and did not
wait for it to start

**Comment by vladzcloudius** [comment]
On 06/29/2016 10:09 AM, Vladislav Zolotarov wrote:
>
>
> On Jun 29, 2016 09:32, "Shlomi Livne" <notifications@github.com 
> <mailto:notifications@github.com>> wrote:
> >
> > I am not sure this is correct
> >
> > [centos@ip-10-181-64-140 ~]$ cat /etc/sysconfig/scylla-server
> > # choose following mode: virtio, dpdk, posix
> > NETWORK_MODE=posix
> >
> > # tap device name(virtio)
> > TAP=tap0
> >
> > # bridge device name (virtio)
> > BRIDGE=virbr0
> >
> > # ethernet device name
> > IFNAME=eth0
> >
> > # setup NIC's interrupts, RPS, XPS (posix)
> > SET_NIC=yes
> >
> > # ethernet device driver (dpdk)
> > ETHDRV=
> >
> > # ethernet device PCI ID (dpdk)
> > ETHPCIID=
> >
> > # number of hugepages
> > NR_HUGEPAGES=64
> >
> > # user for process (must be root for dpdk)
> > USER=scylla
> >
> > # group for process
> > GROUP=scylla
> >
> > # scylla home dir
> > SCYLLA_HOME=/var/lib/scylla
> >
> > # scylla config dir
> > SCYLLA_CONF=/etc/scylla
> >
> > # scylla arguments (for posix mode)
> > SCYLLA_ARGS="--log-to-syslog 1 --log-to-stdout 0 --default-log-level 
> info --collectd-address=127.0.0.1:25826 <http://127.0.0.1:25826> 
> --collectd=1 --collectd-poll-period 3000 --network-stack posix"
> >
> > ## scylla arguments (for dpdk mode)
> > #SCYLLA_ARGS="--log-to-syslog 1 --log-to-stdout 0 
> --default-log-level info --collectd-address=127.0.0.1:25826 
> <http://127.0.0.1:25826> --collectd=1 --collectd-poll-period 3000 
> --network-stack native --dpdk-pmd"
> >
> > # setup as AMI instance
> > AMI=yes
> >
> > this means that scylla_prepare prio to starting scylla will run the 
> posix_net_conf
> >
> > scylla_prepare will run
> >
> >     if [ "$SET_NIC" = "yes" ]; then
> >         NRXQ=`find /sys/class/net/$IFNAME/queues -name "rx-*"|wc -l`
> >         if [ $NRXQ -gt 1 ]; then
> >             CONF_ARGS=-mq
> >         else
> >             CONF_ARGS=-sq
> >         fi
> >         /usr/lib/scylla/posix_net_conf.sh $IFNAME $CONF_ARGS
> >
> > since we get a result of 1 we will run posix_net_conf with -sq
>
> The logic above is bogus: -sq mode should be started not only when the 
> result above is 1 (for Nir it would be 2, by the way). See the 
> corresponding logic in the posix_net_conf.sh script.
>

The simple fix is to remove the whole logic above all together and let 
posix_net_conf.sh decide which mode to configure.

The above logic assumes a single HW queue. Luckily for us this is the 
case (for now) for current in-box ixgbevf drivers in distros we support 
- they all have an antique 2.12.xxx version driver.
However Amazon recommends here 
(http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/sriov-networking.html) 
using to use at least version 2.14.2 and it's not about a number of HW 
queues - it's about bug fixes.

The fact that our instances, that we declare as supporting Enhanced 
Networking, are not built as Amazon instructs is one (really bad) thing. 
The worse thing is that when a user IS going to follow
the Amazon's manual and "fix" the instance built from our AMI, it's 
going to break since the code above will no longer work.

And I don't think that our claim "you shouldn't have changed our AMI" 
would be acceptable in that case.

**Comment by slivne** [comment]
@vladzcloudius I am not arguing bugs - I need to know what users are getting on the current AMI that is already released  and if people are having issue with the existing AMI - not anything else

If anyone installs us on AWS using RPMS its a different issue - we should also make it work good - yet its a different bug.

@nirmaayan I started a 3 node cluster using the AMI and it did setup the networking correctly - (I checked all 3 servers - I am not able to reproduce this - I think that sample you gave is from a system with the enhanced driver vlad is talking about since when I do
```
cat /sys/class/net/eth0/queues/*/rps_cpus
00000000,00000000,00000000,fffefffe
```

I get a single line and you get two.

which means I have a single queue and you have two queues

can you do 

```
 ls /sys/class/net/eth0/queues/*/rps_cpus
/sys/class/net/eth0/queues/rx-0/rps_cpus
```

if you installed the enhanced driver then the scylla_prepare will not work in 1.2 - but its not our AMI.

**Comment by vladzcloudius** [comment]
On 06/29/2016 01:42 PM, Shlomi Livne wrote:
>
> @vladzcloudius <https://github.com/vladzcloudius> I am not arguing 
> bugs - I need to know what users are getting on the current AMI that 
> is already released and if people are having issue with the existing 
> AMI - not anything else
>

I've just described an issue a user would have installing our AMI and 
then do what the Amazon's Enhanced Networking manual instructs it to. 
Note that Amazon's manual describing this flow exactly: first create an 
AMI (they use Ubuntu 14.04 as an example there) and then fix it to work 
properly with the Enhanced Networking.
This would break scylla's networking configuration scripts.

Here is a github issue for a reference: 
https://github.com/scylladb/scylla-ami/issues/16

**Comment by vladzcloudius** [comment]
On 06/29/2016 01:42 PM, Shlomi Livne wrote:
>
> @vladzcloudius <https://github.com/vladzcloudius> I am not arguing 
> bugs - I need to know what users are getting on the current AMI that 
> is already released and if people are having issue with the existing 
> AMI - not anything else
>
> If anyone installs us on AWS using RPMS its a different issue - we 
> should also make it work good - yet its a different bug.
>
> @nirmaayan <https://github.com/nirmaayan> I started a 3 node cluster 
> using the AMI and it did setup the networking correctly - (I checked 
> all 3 servers - I am not able to reproduce this - I think that sample 
> you gave is from a system with the enhanced driver vlad is talking 
> about since when I do
>
> |cat /sys/class/net/eth0/queues/*/rps_cpus 
> 00000000,00000000,00000000,fffefffe |
>
> I get a single line and you get two.
>
> which means I have a single queue and you have two queues
>
> can you do
>
> |ls /sys/class/net/eth0/queues/*/rps_cpus 
> /sys/class/net/eth0/queues/rx-0/rps_cpus |
>
> if you installed the enhanced driver then the scylla_prepare will not 
> work in 1.2 - but its not our AMI.
>

@slivne @nirmaayan <https://github.com/nirmaayan> You can't tell a user 
that he's forbidden to follow Amazon's manual.

**Comment by nirmaayan** [comment]
@tzach can it be that the test and deploy change the setup installation?

**Comment by tzach** [comment]
> @tzach can it be that the test and deploy change the setup installation?

Can you be more specific? which file / value do you suspect ?

**Comment by vladzcloudius** [comment]
@tzach It's in the context of this issue. @nirmaayan is asking a rhetorical question if a user is allowed to modify his/her instance configuration according to Amazon's manual after he/she creates it from our AMI.

**Comment by tzach** [comment]
I was able to reproduce with ami-31b97c51  (Scylla 1.2), i2.8xlarge

```
$ cat cat /sys/class/net/eth0/queues/*/rps_cpus
cat: cat: No such file or directory
00000000,00000000,00000000,00000000
00000000,00000000,00000000,00000000
[centos@ip-172-31-42-61 ~]$ rpm -qa | grep scylla
scylla-conf-1.2.0-20160614.c384b23.el7.centos.x86_64
scylla-boost-thread-1.58.0-11.el7.centos.x86_64
scylla-boost-test-1.58.0-11.el7.centos.x86_64
scylla-tools-1.2.0-20160614.cfd4575.el7.centos.noarch
scylla-boost-system-1.58.0-11.el7.centos.x86_64
scylla-boost-filesystem-1.58.0-11.el7.centos.x86_64
scylla-boost-date-time-1.58.0-11.el7.centos.x86_64
scylla-jmx-1.2.0-20160614.a69d6a9.el7.centos.noarch
scylla-env-1.0-1.el7.centos.noarch
scylla-boost-program-options-1.58.0-11.el7.centos.x86_64
scylla-server-1.2.0-20160614.c384b23.el7.centos.x86_64
```

**Comment by dorlaor** [comment]
On Thu, Jun 30, 2016 at 10:48 AM, Tzach Livyatan <notifications@github.com>
wrote:

> I was able to reproduce with ami-31b97c51 (Scylla 1.2), i2.8xlarge
>
> $ cat cat /sys/class/net/eth0/queues/*/rps_cpus
> cat: cat: No such file or directory
>
>
Are you sure it's with enhanced networking (meaning there is a eth0)?

**Comment by vladzcloudius** [comment]
On 06/30/2016 10:53 AM, Dor Laor wrote:
> On Thu, Jun 30, 2016 at 10:48 AM, Tzach Livyatan 
> <notifications@github.com>
> wrote:
>
> > I was able to reproduce with ami-31b97c51 (Scylla 1.2), i2.8xlarge
> >
> > $ cat cat /sys/class/net/eth0/queues/*/rps_cpus
> > cat: cat: No such file or directory
> >
> >
> Are you sure it's with enhanced networking (meaning there is a eth0)?

The `cat /sys/class/net/*eth0*/queues/*/rps_cpus` implies it... ;)

**Comment by tzach** [comment]
```
$ ifconfig -a
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 9001
        inet 172.31.42.61  netmask 255.255.240.0  broadcast 172.31.47.255
        inet6 fe80::5a:c4ff:fe36:75c9  prefixlen 64  scopeid 0x20<link>
        ether 02:5a:c4:36:75:c9  txqueuelen 1000  (Ethernet)
        RX packets 15422  bytes 12667122 (12.0 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 9132  bytes 949297 (927.0 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 0  (Local Loopback)
        RX packets 315457  bytes 306267202 (292.0 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 315457  bytes 306267202 (292.0 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0```

**Comment by slivne** [comment]
@nirmaayan can you check on 1.3 AMIs if it is ok - lets close this

**Comment by slivne** [comment]
@nirmaayan ping - any update ?

**Comment by nirmaayan** [comment]
@slivne issue still happens
ami-09a63a1e
[centos@ip-172-30-0-51 ~]$  cat /sys/class/net/eth0/queues/*/rps_cpus
00000000,00000000,00000000,00000000
00000000,00000000,00000000,00000000

**Comment by vladzcloudius** [comment]
On 08/11/2016 03:20 PM, Nir wrote:
>
> @slivne <https://github.com/slivne> issue still happens
> ami-09a63a1e
> [centos@ip-172-30-0-51 ~]$ cat /sys/class/net/eth0/queues/*/rps_cpus
> 00000000,00000000,00000000,00000000
> 00000000,00000000,00000000,00000000
>

what type of instance is it?

**Comment by nirmaayan** [comment]
i2.8xlarge

**Comment by syuu1228** [comment]
@nirmaayan @tzach Did you check rps_cpus *AFTER* scylla started?
Since i2.8xlarge has larger ephemeral disks by default, RAID/XFS construction takes times, and posix_net_conf.sh will call only after the construction finishes.
You can check a status of the server by login splash, below of a documentation url. If it constructing RAID/XFS, it says "    Constructing RAID volume...".
How about the result 10 min after starting instances?
Probably the construction finished at that time.

**Comment by nirmaayan** [comment]
@syuu1228 I checked it in couple of stages
1. As soon as I obtain the IP from AWS GUI
2. 20 minutes~ after Scylla was up
Both cases result was the same. 
I will double check again tomorrow with the new version to verify and let you know

**Comment by nirmaayan** [comment]
Double check it with ami-45811a52
1. As soon as I obtain the IP from AWS GUI
2. 20 minutes~ after Scylla was up

![0000](https://cloud.githubusercontent.com/assets/18263247/17688859/c1d284f6-638b-11e6-9bbc-97a107bbd8f3.png)

**Comment by dorlaor** [comment]
How about to run the script with sh -e and see, command by command
what happens when we write a different value there?

You can just echo 0xffffffffffff > THAT_PROC_DIRECTORY

**Comment by syuu1228** [comment]
@nirmaayan okay, then probably it's not related with RAID/XFS construction.
I'll test the script on same instance type.

**Comment by syuu1228** [comment]
I was confused the discussion, and also confused about how posix_net_conf.sh works.
posix_net_conf.sh turns on RPS only when NIC is single queue.
But our AMI turns on enhanced networking, so it's multi queue NIC. Only has two queue though.
So the script does not turns on RPS.

However, if we run posix_net_conf.sh without -mq option, it configures NIC as single queue.
So you can see RPS is configured:
```
$ sudo /usr/lib/scylla/posix_net_conf.sh eth0
00000000,00000000,00000000,00fe00fe
00000000,00000000,00000000,ff00ff00
```

So I think this is all correct.
But we have some problems here, as Vlad mentioned, probably posix_net_conf.sh should detect MQ/SQ without parameter.
Also we have ixgbevf driver version issue, should fixed on ami script, too.

**Comment by vladzcloudius** [comment]
On Aug 17, 2016 01:45, "Takuya ASADA" <notifications@github.com> wrote:
>
> I was confused the discussion, and also confused about how
posix_net_conf.sh works.
> posix_net_conf.sh turns on RPS only when NIC is single queue.
> But our AMI turns on enhanced networking, so it's multi queue NIC. Only
has two queue though.
> So the script does not turns on RPS.
>
> However, if we run posix_net_conf.sh without -mq option, it configures
NIC as single queue.
> So you can see RPS is configured:
>
> $ sudo /usr/lib/scylla/posix_net_conf.sh eth0
00000000,00000000,00000000,00fe00fe 00000000,00000000,00000000,ff00ff00
>
> So I think this is all correct.
> But we have some problems here, as Vlad mentioned, probably
posix_net_conf.sh should detect MQ/SQ without parameter.

@takuya posix_net_conf.sh detects the MQ/SQ by itself already and u have
just showed HOW it works above.
However, if a caller explicitly specifies the mode it would configure
networking according to the specified mode.
The AMI scripts should STOP trying to detect the networking mode (because
it's doong it wrong) and let the script do that itself. Namely the script
should ALWAYS be invoked without -mq/-sq parameters.

> Also we have ixgbevf driver version issue, should fixed on ami script,
too.

**Comment by syuu1228** [comment]
On Wed, Aug 17, 2016 at 12:46 PM, vladzcloudius <notifications@github.com>
wrote:

> On Aug 17, 2016 01:45, "Takuya ASADA" <notifications@github.com> wrote:
> >
> > I was confused the discussion, and also confused about how
> posix_net_conf.sh works.
> > posix_net_conf.sh turns on RPS only when NIC is single queue.
> > But our AMI turns on enhanced networking, so it's multi queue NIC. Only
> has two queue though.
> > So the script does not turns on RPS.
> >
> > However, if we run posix_net_conf.sh without -mq option, it configures
> NIC as single queue.
> > So you can see RPS is configured:
> >
> > $ sudo /usr/lib/scylla/posix_net_conf.sh eth0
> 00000000,00000000,00000000,00fe00fe 00000000,00000000,00000000,ff00ff00
> >
> > So I think this is all correct.
> > But we have some problems here, as Vlad mentioned, probably
> posix_net_conf.sh should detect MQ/SQ without parameter.
>
> @takuya posix_net_conf.sh detects the MQ/SQ by itself already and u have
> just showed HOW it works above.
> However, if a caller explicitly specifies the mode it would configure
> networking according to the specified mode.
> The AMI scripts should STOP trying to detect the networking mode (because
> it's doong it wrong) and let the script do that itself. Namely the script
> should ALWAYS be invoked without -mq/-sq parameters.
>
Okay, will drop detect code on the AMI script.

**Comment by tzach** [comment]
@syuu1228 any updates?

**Comment by syuu1228** [comment]
"[PATCH scylla] dist/common/scripts: drop unnecesarry multiqueue NIC
detection code on scylla_prepare” is the fix.
Also ixgbevf driver update patch will send soon.

**Comment by vladzcloudius** [comment]
On 08/18/2016 11:03 AM, Takuya ASADA wrote:
> "[PATCH scylla] dist/common/scripts: drop unnecesarry multiqueue NIC
> detection code on scylla_prepare” is the fix.

Pls., see my comments on the above patch thread.

**Comment by syuu1228** [comment]
Please close this issue, since this is not a bug.

**Comment by penberg** [comment]
@syuu1228 @vladzcloudius I read through the thread and I don't understand anything.

Can someone summarize what the problem @nirmaayan reported is, why it is not a bug, and what steps we made to ensure nobody will report the issue as a bug in the future?

**Comment by vladzcloudius** [comment]
@penberg @syuu1228 
Takuya, how come it's not a bug?

**Comment by slivne** [comment]
@syuu1228 is this fixed by scylla@533dc0485d5568fdc3f738428aa655e4553d80c1

**Comment by syuu1228** [comment]
Very sorry for confusing the discussion, my previous post and my understand were wrong, THIS IS BUG.
And fixed at 74d994f6a14d52b22db8de5549290e75a1a8f09d.


On @nirmaayan original post, scylla_prepare sets following bits on rps_cpus:
```
$ cat /sys/class/net/eth0/queues/*/rps_cpus
00000000,00000000,00000000,00000000
00000000,00000000,00000000,00000000
```
RPS enabled bit is not set, so this is MQ configuration.

However, running posix_net_conf.sh sets different way:
```
$ sudo /usr/lib/scylla/posix_net_conf.sh eth0
00000000,00000000,00000000,00fe00fe
00000000,00000000,00000000,ff00ff00
```
Now RPS enabled bit is set, this is SQ configuration.
And eth0 on the AMI is ixgbevf, has 2 queues, so it is MQ NIC.

At first I thought using MQ configuration is correct, since the NIC is MQ.
Then posix_net_conf.sh picks SQ configuration due to missing '-mq' option.

But it was misunderstanding, posix_net_conf.sh become able to auto-detect MQ, and it determines eth0 is SQ, since it has too few queues compare to number of cpus.

So real problem is scyll_prepare also has MQ auto-detection code, and doesn't follow posix_net_conf.sh's MQ detection logic.

To fix the problem, following patchset are posted (some of them are already merged to master):
- scylladb/scylla
 - "[PATCH scylla v4 1/2] dist/common/scripts/scylla_prepare: drop unnecesarry multiqueue NIC detection code on scylla_prepare"
   So this fixes #1406.
   It drops scylla_prepare's MQ detection code, use posix_net_conf.sh's decision.
   
- scylladb/seastar
 - "[PATCH seastar] scripts/posix_net_conf.sh: add support --cpu-mask mode"
 - "[PATCH seastar 1/2] scripts/posix_net_conf.sh: fix 'command not found' error when specifies --cpu-mask"
 - "[PATCH seastar 2/2] scripts/posix_net_conf.sh: supress 'ls: cannot access /sys/class/net/<NIC>/device/msi_irqs/' error message"
   These are adding --cpu-mask on posix_net_conf.sh to passes rps cpumask to scylla scripts.



We have one more issues, I already posted the patch and it's already merged, but maybe better to create issue for it.
On SQ mode, posix_net_conf.sh limits packet processing cpus to exclude cpu0 and its sibling.
It's better to exclude cpu0 and its sibling from scylla cpuset, to improve cpu locallity and performance.
So we need to pass rps cpumask to scylla cpuset argument.

Patches to fix this issue:
 - "[PATCH scylla v4 2/2] dist/common/scripts/scylla_sysconfig_setup: sync cpuset parameters with rps_cpus settings when posix_net_conf.sh is enabled and NIC is single queue"
- scylladb/scylla-ami
 - "[PATCH scylla-ami] scylla_ami_setup: remove scylla_cpuset_setup, since we configure it on scylla_sysconfig_setup now"
   We're already setting cpuset statically on AMI, need to drop it and set cpuset based on rps mask.


We need all of them on 1.3.1, also we need to merge scylla-ami patch for master(it's the only patch missing on master).

**Comment by penberg** [comment]
I applied that scylla-ami patch and backported all of the listed patches to 1.3.

@syuu1228 please double-check the branch(es).

**Comment by slivne** [comment]
I checked the ami of the candidate 1.3.1 and I have a single queue

[centos@ip-10-166-155-186 ~]$ cat /sys/class/net/eth0/queues/*/rps_cpus
00000000,00000000,00000000,fffefffe

@syuu1228 is 1.3.1 supposed to include the updated driver - or that will only be part of 1.4

**Comment by syuu1228** [comment]
Not sure this is because driver version, because i able to see two queues
on 1.3.1 AMI (built by myself, confirmed on c3.2xlarge), could be depends
on instance types?
BTW, ixgbevf driver update is not part of 1.3.1.

**Comment by nirmaayan** [comment]
Verify it closed in 1.3
